lib/genesis/helpers.go

```diff
@@ -225,30 +233,32 @@ func buildRawArrayInterface(a []interface{}, kv *keyValue) {
 			// todo check to confirm it's an address
 			tba := crypto.PublicAddressToByteArray(common.Address(v2))
 			kv.value = kv.value + fmt.Sprintf("%x", tba)
+			kv.iVal = append(kv.iVal, tba)
 		case float64:
 			encVal, err := scale.Encode(uint64(v2))
 			if err != nil {
 				//todo determine how to handle this error
 			}
 			kv.value = kv.value + fmt.Sprintf("%x", encVal)
+			kv.iVal = append(kv.iVal, encVal)
 		}
 	}
 }
 
-func formatKey(key []string) (string, error) {
+func formatKey(kv *keyValue) (string, error) {
 	switch true {
-	case reflect.DeepEqual([]string{"grandpa", "authorities"}, key):
+	case reflect.DeepEqual([]string{"grandpa", "authorities"}, kv.key):
 		kb := []byte(`:grandpa_authorities`)
 		return common.BytesToHex(kb), nil
-	case reflect.DeepEqual([]string{"system", "code"}, key):
+	case reflect.DeepEqual([]string{"system", "code"}, kv.key):
 		kb := []byte(`:code`)
 		return common.BytesToHex(kb), nil
 	default:
-		prefix, err := common.Twox128Hash([]byte(key[0]))
+		prefix, err := common.Twox128Hash([]byte(kv.key[0]))
 		if err != nil {
 			return "", err
 		}
-		keydata, err := common.Twox128Hash([]byte(key[1]))
+		keydata, err := common.Twox128Hash([]byte(kv.key[1]))
```

Comment:
Should we have a length check? It might segfault if length is 0 or 1.

Reply:
Yes, good idea.